### PR TITLE
Add a tutorial for data sharing with vineyard on kubernetes via vineyardctl

### PR DESCRIPTION
**Pull request by dashanji**
What do these changes do?
-------------------------

* Add the tutorial for data sharing with vineyard on kubernetes via vineyardctl.
* Add the volumes and volumesMount to workload in the command .
* Support to inject sidecar container to a pod more than a workload.

Related issue number
--------------------

Fixes https://github.com/v6d-io/v6d/issues/1482

### CODE REVIEW

File: docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst [hunk 1]
```diff
@@ -0,0 +1,266 @@
+Data sharing with vineyard on kubernetes 
+========================================
```
Comment (sighingnow) [inline docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst:2]: `Data Sharing with Vineyard on Kubernetes`

File: docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+If you want to share data between different containers or workloads(pods) on kubernetes, it's a good idea to
```
Comment (sighingnow) [inline docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst:4]: between different workloads (pods or containers)

File: docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+on the kubernetes cluster (default is 3 replicas) and the app pods will be scheduled to the vineyard deployment 
+to share the vineyard data via the command `vineyardctl schedule workload`.
+
+Prerequisites
+-------------
+
+- A kubernetes cluster with version >= 1.25.10.
+- Install the latest vineyardctl command line tool refer to `vineyardctl installation`_.
+
+
```
Comment (sighingnow) [inline docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst:33]: No extra blank lines.

File: docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: vineyard-showcase
+      namespace: vineyard-test
+    spec:
+      containers:
+      - name: producer
+        image: python:3.10
+        command: ["/bin/sh", "-c", "pip install --user -q --disable-pip-version-check vineyard numpy pandas; python -c \"$put_dataframe\"; python -c \"$put_basic_data_unit\"; sleep infinity"]
```
Comment (sighingnow) [inline docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst:55]: Use:

```
command:
- bash
- -c
- |
  pip install ....
  
  echo >> producer.py <EOF
  import vineyard..
  ...
  ...
  EOF

  python producer.py
```

File: docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+    # get the logs of the consumer container
+    $ kubectl logs -f vineyard-showcase -n test -c consumer
+    A   -30.168469
+    B   -19.269489
+    C     6.332533
+    D    -9.714950
+    dtype: float64
+    (1, 1.2345000505447388, 'xxxxabcd')
+
+
```
Comment (sighingnow) [inline docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst:118]: No extra blank lines.

File: docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+from vineyard and do some computation.
+
+Save the following yaml as `pod.yaml`.
+
+.. code:: yaml
+
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: vineyard-showcase
```
Comment (sighingnow) [inline docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst:49]: vineyard-showcase -> vineyard-producer-consumer

File: docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+    service/vineyard-sidecar-rpc created
+    pod/vineyard-showcase created
+
+
+Then you can get the logs of the consumer containers as follows.
+
+.. code:: bash
+
+    # get the logs of the consumer container
+    $ kubectl logs -f vineyard-showcase -n test -c consumer
```
Comment (sighingnow) [inline docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst:110]: vineyard-showcase -> vineyard-producer-consumer

File: docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+    A   -30.168469
+    B   -19.269489
+    C     6.332533
+    D    -9.714950
+    dtype: float64
+    (1, 1.2345000505447388, 'xxxxabcd')
+
+
+Data sharing between different workloads
+----------------------------------------
```
Comment (sighingnow) [inline docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst:120]: different workloads -> different pods

Comment (sighingnow) [reply docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst:120]: Producer and consumer are different workloads, even when they are placed in the same pod.

**Review by sighingnow — CHANGES_REQUESTED** [review]
Another minor comment: It's good to mix code change **with its documentation** in the same PR. But it is not so good to mix code changes with a pull request about "adding new tutorials".

**Comment by dashanji** [comment]
> #1482

Make sense to me. I'll split the code changes into another pr.

**Comment by dashanji** [comment]
> Another minor comment: It's good to mix code change **with its documentation** in the same PR. But it is not so good to mix code changes with a pull request about "adding new tutorials".

Make sense to me. I'll split the code changes into another pr.

File: docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+              pip install vineyard numpy pandas > /dev/null 2>&1;
+              cat << EOF >> producer.py
+              import vineyard
+              import numpy as np
+              import pandas as pd
+              client = vineyard.connect()
+              client.put(pd.DataFrame(np.random.randn(100, 4), columns=list('ABCD')), persist=True, name="test_dataframe")
+              client.put((1, 1.2345, 'xxxxabcd'), persist=True, name="test_basic_data_unit");
+              client.close()
+              EOF
```
Comment (sighingnow) [inline docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst:191]: Leave a blank line between groups of commands would be more clear.

File: docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+            app: producer
+        spec:
+          containers:
+          - name: producer
+            image: python:3.10
+            command:
+            - bash
+            - -c
+            - |
+              pip install vineyard numpy pandas > /dev/null 2>&1;
```
Comment (sighingnow) [inline docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst:182]: Its ok to leave it logging.

File: docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+    EOF
+
+Use the `vineyardctl` to schedule the two workloads on the vineyard cluster.
+
+.. code:: bash
+
+    # schedule the producer workload to the vineyard cluster and apply it to the kubernetes cluster
+    $ vineyardctl schedule workload -f producer.yaml --vineyardd-name vineyardd-sample \
+    --vineyardd-namespace vineyard-test -o yaml | kubectl apply -f -
+    deployment.apps/producer created
```
Comment (sighingnow) [inline docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst:244]: For multiple-line commands, leave proper indentation, e.g., 


```
.. code:: bash
    # schedule the producer workload to the vineyard cluster and apply it to the kubernetes cluster
    $ vineyardctl schedule workload -f producer.yaml --vineyardd-name vineyardd-sample \
          --vineyardd-namespace vineyard-test -o yaml | kubectl apply -f -
          deployment.apps/producer created
```

Would be more readable.

**Review by sighingnow — APPROVED** [review]
Rebase to main.

File: docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst (hunk 11 above)
Comment (dashanji) [reply docs/tutorials/kubernetes/data-sharing-with-vineyard-on-kubernetes.rst:244]: Get it. Thanks!

**Pull request merged**